challenges-14-plus.yml

```diff
@@ -0,0 +1,472 @@
+  question: |
+    In order to fix this page transition issue (and save our jobs), we need to find
+    a way to identify when a `turbo:before-render` or a `turbo:render` event come
+    from a preview or a real page.
+    
+    We need to make sure our previews have the `turbo-loading` class in their `<body>`
+    elements so that the red doesn't flash twice!
+    
+    How can we do this?
+  explanation: |
+    The `data-turbo-preview` attribute added to the `<html>` element is very useful in
+    cases in which we need to identify wheather we're dealing with a preview or a real
+    page in Turbo.
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          You can do this by performing this check using JavaScript:
+  
+          ```javascript
+          document.body.hasAttribute('data-turbo-preview');
+          ```
+        b: |
+          You can do this by performing this check using JavaScript:
+  
+          ```javascript
+          document.querySelector('body').hasAttribute('data-turbo-preview');
+          ```
+        c: |
+          You can do this by performing this check using JavaScript:
+          
+          ```javascript
+          document.documentElement.hasAttribute('data-turbo-preview');
+          ```
+        d: |
+          There is no way of differenciating between a preview and a real page in
+          Turbo.
+      correct: c
+
+native_prefetch:
+  question: |
+    Let's say our website has a very important landing page, a route by the name
+    `app_pizzas` that we want to prefetch on every page so that such page is
+    always ready to be loaded instantly if the user wants to go there.
+    
+    Assuming we have a `{% block metas %}{% endblock %}` that we can override. What
+    is the native way in which we can do this?
+  explanation: |
+    `<link rel="prefetch">` is a useful tool that you can leverage with Turbo Drive
+    when you need to make certain pages readily available to the user.
+    
+    In our example, we infer that we are adding this prefetch link on every page. Make
+    sure you have this in mind, because if you add a prefetch link in a *specific* page,
+    this won't stay in the `<head>` element for other pages!
+  features:
+    multiple_choice:
+      a: |
+        ```twig
+        <link rel="prefetch" src="{{ path('app_pizzas') }}">
+        ```
+      b: |
+        ```twig
+        <link rel="prefetch" href="{{ path('app_pizzas') }}">
+        ```
+      c: |
+        ```twig
+        <a rel="prefetch" href="{{ path('app_pizzas') }}"></a>
+        ```
+      d: |
+        ```twig
+        <meta rel="prefetch" href="{{ path('app_pizzas') }}">
+        ```
+    correct: b
+
+lazy_frame:
+  question: |
+    What is a Turbo Frame and what does it do?
+  explanation: |
+    The `<turbo-frame>` HTML element is a special element that allows us
+    to make areas of our web pages dynamic.
+    
+    You can use Turbo Frames if you want navigation to happen inside an
```

Comment:
```suggestion
    You can use Turbo Frames if you want navigation (e.g. link clicks and form submits) to
    happen (or "stay") inside of a specific area, instead of affecting the entire page.
```